iothub/device/src/ClientFactory.cs

```diff
@@ -75,7 +75,7 @@ internal static InternalClient Create(string hostName, IAuthenticationMethod aut
                         if (Logging.IsEnabled)
                             Logging.Error(null, $"{nameof(CertificateInstaller)} failed to read or write to cert store due to: {ex}");
 
-                        throw new UnauthorizedException($"Failed to provide certificates in the chain - {ex.Message}", ex);
+                        throw new IotHubClientException($"Failed to provide certificates in the chain - {ex.Message}", ex, false, IotHubStatusCode.Unauthorized);
```

Comment:
To review an initialization of `IotHubClientException` in this PR, it would be better to check how was the old ctor in the particular exception class (`UnauthorizedException` in this example) [defined](https://github.com/Azure/azure-iot-sdk-csharp/blob/aa4e8a65f3c410e2cc60d1ba80e33213103529bd/iothub/device/src/Exceptions/UnauthorizedException.cs#L41). 

In the additional ctor added for `IotHubClientException` in this PR, I also pass `StatusCode` in.